src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -348,9 +352,13 @@ abstract class UnCurry extends InfoTransform
     /** If an eligible method is entirely wrapped in a call to synchronized
      *  locked on the same instance, remove the synchronized scaffolding and
      *  mark the method symbol SYNCHRONIZED for bytecode generation.
+     *
+     *  Delambdafy targets are deemed ineligible as the Delambdafy phase will
+     *  replace `this.synchronized` with `$this.synchronzed` now that it emits
+     *  all lambda impl methods as static.
      */
     private def translateSynchronized(tree: Tree) = tree match {
-      case dd @ DefDef(_, _, _, _, _, Apply(fn, body :: Nil)) if isSelfSynchronized(dd) =>
+      case dd @ DefDef(_, _, _, _, _, Apply(fn, body :: Nil)) if isSelfSynchronized(dd) && dd.symbol.isDelambdafyFunction =>
```

Comment:
missing a `!` ? and maybe you could add the condition into `isSelfSynchronized`, which already performs some additional checks.